src/Controls/src/SourceGen/RouteSourceGenerator.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Linq;
+using System.Text;
+
+using Microsoft.CodeAnalysis;
+
+using static Microsoft.Maui.Controls.SourceGen.Helper;
+
+namespace Microsoft.Maui.Controls.SourceGen
+{
+	[Generator(LanguageNames.CSharp)]
+	public class RouteSourceGenerator : ISourceGenerator
```

Comment:
This needs to implement `IIncrementalGenerator`, otherwise it will have an impact on incremental build times.

Some details here on this: https://github.com/dotnet/maui/commit/dc9ac94692b321acd87bafec69366a9cd18caddc

Reply:
> This needs to implement `IIncrementalGenerator`, otherwise it will have an impact on incremental build times.

Yes, @pictos informed the same. Working on it. Will update once done and thanks for the reference link.

Reply:
Have migrated to Incremental source generator.

For time being have commented out the `Generator` attribute on the classic option to facilitate the review. Once the review is done, we can remove the following files related to the classic option.

* RouteSourceGenerator.cs
* RouteSyntaxReceiver.cs

Reply:
To be honest, there is a lot going on here. Can we start with a new issue that is a "proposal"?

We should iron out what the design is for this before writing code.

Reply:
> To be honest, there is a lot going on here. Can we start with a new issue that is a "proposal"?

@PureWeen has already opened a [Discussion](https://github.com/CommunityToolkit/Maui/discussions/1004) thread for this in the Maui CommunityToolkit repo.

Reply:
Will update this PR description over there too.

Reply:
https://github.com/CommunityToolkit/Maui/discussions/1004#discussioncomment-5247116